uint8 status
uint8 desired_routine
 